src/coffea/dataset_tools/preprocess.py

```diff
@@ -19,6 +23,7 @@ def _get_steps(
     recalculate_seen_steps: bool = False,
     skip_bad_files: bool = False,
     file_exceptions: Exception | Warning | tuple[Exception | Warning] = (OSError,),
+    calculate_form: bool = False,
```

Comment:
This is more of a comment for the earlier PR but given the thorough docstring here, do we want to promote `get_steps` to be public?

Reply:
On this PR though, I feel like `cache_form` might be more clear?

Reply:
It's not really a cache until it hits the Fileset spec, so I think `save_form` is more accurate?